src/cuckoo.c

```diff
@@ -180,8 +180,8 @@ uint64_t CuckooFilter_Count(const CuckooFilter *filter, CuckooHash hash) {
 int CuckooFilter_Delete(CuckooFilter *filter, CuckooHash hash) {
     LookupParams params;
     getLookupParams(hash, &params);
-    for (uint16_t ii = 0; ii < filter->numFilters; ++ii) {
-        if (Filter_Delete(&filter->filters[ii], &params)) {
+    for (uint16_t ii = filter->numFilters; ii > 0; --ii) {
+        if (Filter_Delete(&filter->filters[ii - 1], &params)) {
```

Comment:
Why make this change?
The idea here is to clear the latest filter which is likely to be the one with the least items in it.

Reply:
My suspicion is this change just hides the real issue here.

Reply:
> My suspicion is this change just hides the real issue here.

In the original code, insertion is bottom up but deletion is top down. 
This miss match can cause the following anomaly:

suppose we insert the following keys a c and
suppose fingerprint(a) == fingerprint(c) and
hash(c) % filter0.numBucket == hash(a) % filter0.numBuckets
hash(c) % filter1.numBucket != hash(a) % filter1.numBuckets

a resided in level0 filter
c reside in level1 filter

filter0: a 
fitler1: c

now we try to delete c from the filter, and since deletion is performed top down; so 'a' is deleted *by mistake*.  
then we try to delete a from the filter, now we cannot find 'a' in fitler1 (it's already deleted from fitler0);
so as long the an expansion of rate > 1 happens and the insertion and deletion order mismatch, deletion may fail.

Reply:
@qingping209 
You are right! We have to delete hashes from the larger to the smaller. It was not apparent before since sizes for all filters were identical.
Thank you very much for this contribution.